### ISSUE TITLE
GPT4 Support

### ISSUE THREAD

**Issue by zachriggle** [Aesthetikx/openai_pipe#6]
It would be nice to be able to specify GPT4 for the model

**Comment by Aesthetikx** [comment]
Agreed, I believe I will have some time to update in the next few days.

**Comment by Aesthetikx** [comment]
@zachriggle  It looks like GPT-4 is still in a private beta, but I will keep my eye on this. Do you have access to the beta? For now, the list of models available to me (at least in so far as what https://github.com/alexrudall/ruby-openai provides) does not yet include GPT-4.

**Comment by slacksystem** [comment]
openai_pipe version 0.1.3 installed fresh from gem
Anything I query I get the error `Unable to fetch response: The model `text-davinci-003` has been deprecated, learn more here: https://platform.openai.com/docs/deprecations`

I see that this repo was updated 2 weeks ago but from what I can tell the current version in the master branch is also 0.1.3. Is this no longer functional or am I doing something wrong. If not, can the model be updated to gpt-4 or at least gpt-3.5-turbo? Better yet, can the model be configurable? Perhaps through an environment variable? That would be ideal. Just confused as this issue has been open almost a year and at least gpt-3.5 has been favored for quite a while. I don't know ruby or I would try to upgrade the model and make a PR.

**Comment by Aesthetikx** [comment]
> openai_pipe version 0.1.3 installed fresh from gem Anything I query I get the error `Unable to fetch response: The model `text-davinci-003` has been deprecated, learn more here: https://platform.openai.com/docs/deprecations`
> 
> I see that this repo was updated 2 weeks ago but from what I can tell the current version in the master branch is also 0.1.3. Is this no longer functional or am I doing something wrong. If not, can the model be updated to gpt-4 or at least gpt-3.5-turbo? Better yet, can the model be configurable? Perhaps through an environment variable? That would be ideal. Just confused as this issue has been open almost a year and at least gpt-3.5 has been favored for quite a while. I don't know ruby or I would try to upgrade the model and make a PR.

I will take a look at this, in the mean time try `gem update quick_openai`

**Comment by anvaka** [comment]
Hey @Aesthetikx , I don't know you, nor do I know @zachriggle . I was just randomly exploring github today, found pwndbg project and contributors there [say](https://github.com/pwndbg/pwndbg/blob/da297308b603caee01044d291f99b6b8f194c494/README.md?plain=1#L92):

> Pwndbg was originally created by [Zach Riggle](https://github.com/zachriggle), who is no longer with us

This got me to the @zachriggle's page, and I discovered he passed in May 2023: https://www.smithfamilyfuneralhomes.com/obituaries/zachariah-riggle . 

This issue here seems to be his last contribution to the open source. I just wanted to say thanks to @zachriggle, and I hope he's in better place. 

As long as we remember other people, I believe they are alive. Sending everyone who reads this care and positive wishes.
